packages/enketo-express/public/js/src/module/controller-webform.js

```diff
@@ -443,9 +444,28 @@ function _submitRecord(survey) {
                     );
                 }, 1200);
             } else {
-                msg = msg.length > 0 ? msg : t('alert.submissionsuccess.msg');
-                gui.alert(msg, t('alert.submissionsuccess.heading'), level);
-                _resetForm(survey);
+                const parser = new DOMParser();
+                const xmlForm = parser.parseFromString(
+                    data.modelStr,
+                    'text/xml'
+                );
+                const maybeSubmitMessage =
+                    xmlForm.querySelector('submitMessage');
+                msg = maybeSubmitMessage
+                    ? result.message.length > 0
+                        ? jstransformer.render(result.message)
```

Comment:
Any XSS opportunities?

Reply:
Great point. I'm looking around enketo for a markdown sanitization utility. Any pointers?